// @dasha.ai/sdk/dist/grpc-api/node/generated/nlu-service_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!
'use strict';
var grpc = require('@grpc/grpc-js');
var nlu$service_pb = require('./nlu-service_pb.js');
function serialize_dasha_sdk_AvailableSkillsRequest(arg) {
    if (!(arg instanceof nlu$service_pb.AvailableSkillsRequest)) {
        throw new Error('Expected argument of type dasha.sdk.AvailableSkillsRequest');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_dasha_sdk_AvailableSkillsRequest(buffer_arg) {
    return nlu$service_pb.AvailableSkillsRequest.deserializeBinary(new Uint8Array(buffer_arg));
}
function serialize_dasha_sdk_AvailableSkillsResponse(arg) {
    if (!(arg instanceof nlu$service_pb.AvailableSkillsResponse)) {
        throw new Error('Expected argument of type dasha.sdk.AvailableSkillsResponse');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_dasha_sdk_AvailableSkillsResponse(buffer_arg) {
    return nlu$service_pb.AvailableSkillsResponse.deserializeBinary(new Uint8Array(buffer_arg));
}
function serialize_dasha_sdk_RecognitionRequest(arg) {
    if (!(arg instanceof nlu$service_pb.RecognitionRequest)) {
        throw new Error('Expected argument of type dasha.sdk.RecognitionRequest');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_dasha_sdk_RecognitionRequest(buffer_arg) {
    return nlu$service_pb.RecognitionRequest.deserializeBinary(new Uint8Array(buffer_arg));
}
function serialize_dasha_sdk_RecognitionResponse(arg) {
    if (!(arg instanceof nlu$service_pb.RecognitionResponse)) {
        throw new Error('Expected argument of type dasha.sdk.RecognitionResponse');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_dasha_sdk_RecognitionResponse(buffer_arg) {
    return nlu$service_pb.RecognitionResponse.deserializeBinary(new Uint8Array(buffer_arg));
}
function serialize_dasha_sdk_TrainRequest(arg) {
    if (!(arg instanceof nlu$service_pb.TrainRequest)) {
        throw new Error('Expected argument of type dasha.sdk.TrainRequest');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_dasha_sdk_TrainRequest(buffer_arg) {
    return nlu$service_pb.TrainRequest.deserializeBinary(new Uint8Array(buffer_arg));
}
function serialize_dasha_sdk_TrainStatusStream(arg) {
    if (!(arg instanceof nlu$service_pb.TrainStatusStream)) {
        throw new Error('Expected argument of type dasha.sdk.TrainStatusStream');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_dasha_sdk_TrainStatusStream(buffer_arg) {
    return nlu$service_pb.TrainStatusStream.deserializeBinary(new Uint8Array(buffer_arg));
}
var NluServiceService = exports.NluServiceService = {
    train: {
        path: '/dasha.sdk.NluService/Train',
        requestStream: false,
        responseStream: true,
        requestType: nlu$service_pb.TrainRequest,
        responseType: nlu$service_pb.TrainStatusStream,
        requestSerialize: serialize_dasha_sdk_TrainRequest,
        requestDeserialize: deserialize_dasha_sdk_TrainRequest,
        responseSerialize: serialize_dasha_sdk_TrainStatusStream,
        responseDeserialize: deserialize_dasha_sdk_TrainStatusStream,
    },
    recognize: {
        path: '/dasha.sdk.NluService/Recognize',
        requestStream: false,
        responseStream: false,
        requestType: nlu$service_pb.RecognitionRequest,
        responseType: nlu$service_pb.RecognitionResponse,
        requestSerialize: serialize_dasha_sdk_RecognitionRequest,
        requestDeserialize: deserialize_dasha_sdk_RecognitionRequest,
        responseSerialize: serialize_dasha_sdk_RecognitionResponse,
        responseDeserialize: deserialize_dasha_sdk_RecognitionResponse,
    },
    getAvailableSkills: {
        path: '/dasha.sdk.NluService/GetAvailableSkills',
        requestStream: false,
        responseStream: false,
        requestType: nlu$service_pb.AvailableSkillsRequest,
        responseType: nlu$service_pb.AvailableSkillsResponse,
        requestSerialize: serialize_dasha_sdk_AvailableSkillsRequest,
        requestDeserialize: deserialize_dasha_sdk_AvailableSkillsRequest,
        responseSerialize: serialize_dasha_sdk_AvailableSkillsResponse,
        responseDeserialize: deserialize_dasha_sdk_AvailableSkillsResponse,
    },
};
exports.NluServiceClient = grpc.makeGenericClientConstructor(NluServiceService);
